extensions/hal/runtime/src/main/java/io/quarkus/hal/HalService.java

```diff
@@ -0,0 +1,79 @@
+package io.quarkus.hal;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Service with Hal utilities. This service is used by the Resteasy Links, Resteasy Reactive Links and the
+ * Rest Data Panache extensions.
+ */
+@SuppressWarnings("unused")
+public abstract class HalService {
+
+    private static final String SELF_REF = "self";
+
+    /**
+     * Wrap a collection of objects into a Hal collection wrapper by resolving the Hal links.
+     * The Hal collection wrapper is then serialized by either json or jackson.
+     *
+     * @param collection The collection of objects to wrap.
+     * @param collectionName The name that will include the collection of objects within the `_embedded` Hal object.
+     * @param entityClass The class of the objects in the collection. If null, it will not resolve the links for these objects.
+     * @return The Hal collection wrapper instance.
+     */
+    public HalCollectionWrapper toHalCollectionWrapper(Collection<Object> collection, String collectionName,
+            Class<?> entityClass) {
+        List<HalEntityWrapper> items = collection.stream()
+                .map(this::toHalWrapper)
```

Comment:
Can we please eliminate the use of lambdas in runtime code?

Reply:
Will delete it. This is a copy/paste of what we were already using in the panache extension and I didn't want to introduce more changes.